### ISSUE TITLE
Remove bce_with_logits in derivatives.yaml

### ISSUE THREAD

**Issue by ailzhang** [pytorch/pytorch#28859]
Its backward is done through autograd, we don't need explicitly set it at all.

**Comment by zou3519** [comment]
The implemented backward functions for bce_with_logits may be more efficient than a regular autograd backward. For instance, we have fused operations like mse_loss that (for CUDA inputs) only launches a single (CUDA) kernel in backward. What problem are you trying to solve by removing bce_with_logits from derivatives.yaml?

**Comment by ailzhang** [comment]
@zou3519 Yea I remember I did this one about 1 yr ago :P https://github.com/pytorch/pytorch/blob/be3808d3b1164a0073f85fc0a58f2c7c27b2ae40/aten/src/ATen/native/Loss.cpp#L122 
I don't recall doing it for a particular reason (maybe I forgot :P ). I just want to double check and remove the redundancy if it was a mistake I made. :D
